src/game.cpp

```diff
@@ -7141,6 +7141,16 @@ void game::smash()
             return; // don't smash terrain if we've smashed a corpse
         }
     }
+    bool ter_or_furn = m.has_flag_ter_or_furn( "ELECTRIFIED", smashx, smashy );
+    if (ter_or_furn) {
+            body_part bp = random_body_part();
```

Comment:
I usually don't smash things with my head or my torso, but that's just me (-:

You should check for the wielded (and therefor used) item (`item::conductive()`). The character should not be damaged when bashing with a suitable (non-conductive) weapon, similar to the code in `mdefense::zapback`.